### PR TITLE
[RHDM-766] Complete KieServer API for PMML

### PR DESCRIPTION
**Pull request by jiripetrlik**
Repaired PR 1715.

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/filtered-resources/kjars-sources/decision-tree/pom.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.kie</groupId>
+        <artifactId>kie-maven-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <usesPMML>true</usesPMML>
+          <properties>
+            <drools.dialect.java.compiler.lnglevel>1.6</drools.dialect.java.compiler.lnglevel>
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/filtered-resources/kjars-sources/decision-tree/pom.xml]: AFAIK this is not needed any more.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/filtered-resources/kjars-sources/scorecard/pom.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.kie</groupId>
+        <artifactId>kie-maven-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <usesPMML>true</usesPMML>
+          <properties>
+            <drools.dialect.java.compiler.lnglevel>1.6</drools.dialect.java.compiler.lnglevel>
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/filtered-resources/kjars-sources/scorecard/pom.xml]: Same as above

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/ApplyDecisionTreeModelIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/decision-tree");
+
+        kjarClassLoader = KieServices.Factory.get().newKieContainer(releaseId).getClassLoader();
+
+        createContainer(CONTAINER_ID, releaseId);
+    }
+
+    @Override
+    protected void addExtraCustomClasses(Map<String, Class<?>> extraClasses) throws Exception {
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/ApplyDecisionTreeModelIntegrationTest.java]: Can be removed from here.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/ApplyScorecardModelIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/scorecard");
+
+        kjarClassLoader = KieServices.Factory.get().newKieContainer(releaseId).getClassLoader();
+
+        createContainer(CONTAINER_ID, releaseId);
+    }
+
+    @Override
+    protected void addExtraCustomClasses(Map<String, Class<?>> extraClasses) throws Exception {
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/ApplyScorecardModelIntegrationTest.java]: Same as above.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/PMMLApplyModelBaseTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    protected RuleServicesClient rulesClient;
+
+    @BeforeClass
+    public static void setupFactory() throws Exception {
+        commandsFactory = KieServices.Factory.get().getCommands();
+    }
+
+    protected RuleServicesClient createRuleServicesClient() throws Exception {
+        KieServicesConfiguration cfg = createKieServicesRestConfiguration();
+        this.rulesClient = createDefaultClient(cfg, MarshallingFormat.JAXB).getServicesClient(RuleServicesClient.class);
```
Comment (sutaakar) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/PMMLApplyModelBaseTest.java]: Is there a reason why you test just REST and JAXB? What about JMS and other marshalling formats?

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/rest/RestURI.java [hunk 6]
```diff
@@ -227,6 +227,9 @@
 
     // DMN URI
     public static final String DMN_URI = "containers/{" + CONTAINER_ID + "}/dmn";
+    
+    // PMML URI
+    public static final String PMML_URI = "containers/{" + CONTAINER_ID + "}/pmml";
```
Comment (sutaakar) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/rest/RestURI.java]: It doesn't seem to be used anywhere.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/PMMLApplyModelBaseTest.java (hunk 5 above)
Comment (lanceleverich) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/PMMLApplyModelBaseTest.java]: I added REST and JAXB tests as simple proofs of the ability to interact with the PMML when it is operating in a kie-server. I can add more tests, but my initial goal was to demonstrate that the API is available and able to be used.

Comment (jiripetrlik) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-drools/src/test/java/org/kie/server/integrationtests/drools/pmml/PMMLApplyModelBaseTest.java]: @sutaakar @lanceleverich 
Tests were updated to use all 3 types of marshallers. However, it seems that Xstream and JSON does not work.

**Comment by jiripetrlik** [comment]
@sutaakar 
Thank you for your review! I've tried to fix all your comments.

**Pull request merged**